### PR TITLE
Fix/leak

### LINKED ISSUE #431: 🐞 Investigate and resolve precompute related leak
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Current Behavior

During precompute process, memory is leaking. Find source and resolve.

### Expected Behavior

_No response_

### Steps To Reproduce

_No response_

### Environment

```markdown
- OS:
```

### Anything else?

_No response_

### PR DESCRIPTION
**Pull request by SteveMaier-IRT**
[//]: # (🚨 Please review the CONTRIBUTING.md in this repository. 💔Thank you!)

### Issue
Fixes #431

### Description
Fixed multiple memory leaks

### Testing
*Describe the best way to test or validate your PR.*

### CODE REVIEW

File: libs/hacl/CMakeLists.txt [hunk 1]
```diff
@@ -21,7 +21,7 @@ CPMAddPackage(
     NAME hacl
     GITHUB_REPOSITORY addressxception/hacl-packages
     VERSION 0.6.0
-    GIT_TAG 8b17413ea5731de2fec5c511f24d8bf85ffcf3e7
+    GIT_TAG 04883d6ce3384ee5c8c31db907ca556ed8b45571
```
Comment (john-s-morgan) [inline libs/hacl/CMakeLists.txt:24]: 🙌🏼  nice catch

File: Makefile [hunk 2]
```diff
@@ -625,9 +625,9 @@ endif
 
 test-netstandard: build-netstandard
 	@echo 🧪 TEST NETSTANDARD $(PROCESSOR) $(TARGET)
-	dotnet test -a $(PROCESSOR) -c $(TARGET) ./bindings/netstandard/ElectionGuard/ElectionGuard.ElectionSetup.Tests/ElectionGuard.ElectionSetup.Tests.csproj
-	dotnet test -a $(PROCESSOR) -c $(TARGET) ./bindings/netstandard/ElectionGuard/ElectionGuard.Encryption.Tests/ElectionGuard.Encryption.Tests.csproj
-	dotnet test -a $(PROCESSOR) -c $(TARGET) ./bindings/netstandard/ElectionGuard/ElectionGuard.Decryption.Tests/ElectionGuard.Decryption.Tests.csproj
+#	dotnet test -a $(PROCESSOR) -c $(TARGET) ./bindings/netstandard/ElectionGuard/ElectionGuard.ElectionSetup.Tests/ElectionGuard.ElectionSetup.Tests.csproj
+	dotnet test -a $(PROCESSOR) -c $(TARGET) --filter TestCategory=MemoryLeak ./bindings/netstandard/ElectionGuard/ElectionGuard.Encryption.Tests/ElectionGuard.Encryption.Tests.csproj
+#	dotnet test -a $(PROCESSOR) -c $(TARGET) ./bindings/netstandard/ElectionGuard/ElectionGuard.Decryption.Tests/ElectionGuard.Decryption.Tests.csproj
```
Comment (john-s-morgan) [inline Makefile:630]: This should be reverted

File: src/electionguard/lookup_table.hpp [hunk 3]
```diff
@@ -176,7 +176,7 @@ namespace electionguard
         AsyncSemaphore task_lock;
         std::map<std::string, std::unique_ptr<LookupTableType>> key_map;
 
-        LookupTableType *getBaseLookupTable(const std::string key, uint64_t (&base)[MAX_P_LEN])
+        LookupTableType *getBaseLookupTable(const std::string &key, uint64_t (&base)[MAX_P_LEN])
```
Comment (john-s-morgan) [inline src/electionguard/lookup_table.hpp:179]: Good catch on this @AddressXception

File: bindings/netstandard/ElectionGuard/ElectionGuard.Encryption.Tests/TestPrecompute.cs [hunk 4]
```diff
@@ -45,37 +49,82 @@ public void Test_Precompute_Status_NoStarted()
...
+        // [Test, Order(4)]
+        // public void Test_Precompute_Using_Old_Interface()
+        // {
+        //     Console.WriteLine($"Test_Precompute_Using_Old_Interface");
+
+        //     var compute = new Precompute();
+        //     compute.StartPrecomputeAsync(_precompute.PublicKey, TestBufferSize);
+        //     RunFor(TimeSpan.FromSeconds(TestRunLengthInS));
+        //     compute.StopPrecompute();
+        // }
```
Comment (john-s-morgan) [inline bindings/netstandard/ElectionGuard/ElectionGuard.Encryption.Tests/TestPrecompute.cs:99]: cleanup needed. This should be uncommented

File: bindings/netstandard/ElectionGuard/ElectionGuard.Encryption.Tests/TestEncrypt.cs [hunk 5]
```diff
@@ -383,10 +385,42 @@ public void Test_EncryptMediator_Hashes_Dont_Match()
...
+            Console.WriteLine($"Memory Size: {workingSet / (1024.0 * 1024.0):F2} MB");
+        }
+
 
         [Test]
+        [Category("MemoryLeak")]
         public void Test_Multiple_Submit_Check_10k_Records_For_Memory_Leaks()
         {
+            Console.WriteLine($"Test_Multiple_Submit_Check_10k_Records_For_Memory_Leaks");
+
```
Comment (john-s-morgan) [inline bindings/netstandard/ElectionGuard/ElectionGuard.Encryption.Tests/TestEncrypt.cs:423]: Cleanup

**Review by john-s-morgan — CHANGES_REQUESTED** [review]

File: Makefile (hunk 2 above)
Comment (SteveMaier-IRT) [reply Makefile]: fixed

**Review by john-s-morgan — APPROVED** [review]

**Pull request merged**